pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -173,62 +178,58 @@ func (r *Reconciler) DetectOperation(ctx context.Context, log logr.Logger, tinke
 	}
 
 	wantVersionChange := currentKCP.Spec.Version != tinkerbellScope.ControlPlane.KubeadmControlPlane.Spec.Version
-	cpWantScaleChange := *currentKCP.Spec.Replicas != *tinkerbellScope.ControlPlane.KubeadmControlPlane.Spec.Replicas
-	workerWantScaleChange, err := r.WorkerReplicasDiff(ctx, tinkerbellScope)
-	if err != nil {
-		return "", err
-	}
-	wantScaleChange := workerWantScaleChange || cpWantScaleChange
 
 	// The restriction that not allowing scaling and rolling is covered in webhook.
 	op := NoChange
-	switch {
-	case wantScaleChange:
-		op = ScaleOperation
-	case wantVersionChange:
+	if wantVersionChange {
```

Comment:
can you check version change directly in the if condition?
```suggestion
	if currentKCP.Spec.Version != tinkerbellScope.ControlPlane.KubeadmControlPlane.Spec.Version {
```